packages/api-explorer/src/components/ExploreType/ExploreProperty.tsx

```diff
@@ -0,0 +1,262 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2020 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+
+import React, { FC } from 'react'
+import {
+  FlexItem,
+  IconNames,
+  Icon,
+  Tree,
+  TreeItem,
+  Tooltip,
+  Space,
+} from '@looker/components'
+import { IProperty } from '@looker/sdk-codegen'
+import ReactMarkdown from 'react-markdown'
+import {
+  expandable,
+  ExploreTypeLink,
+  pickType,
+  pickTypeProps,
+  typeIcon,
+} from '.'
+
+interface TipIconProps {
+  show: boolean
+  tip: string
+  icon: IconNames
+  title: string
+}
+
+/**
+ * Show a tip with icon if show is true
+ * @param show true to show tip and icon
+ * @param tip to display on hover
+ * @param icon name to use
+ * @param title for icon
+ * @constructor
+ */
+export const TipIcon: FC<TipIconProps> = ({ show, tip, icon, title }) => {
+  if (!show) return <></>
+  return (
+    <Tooltip content={tip}>
+      <Icon name={icon} size="xsmall" content={tip} title={title} />
+    </Tooltip>
+  )
+}
+
+/**
+ * Interface shared by several ExploreProperty components
+ */
+interface ExplorePropertyProps {
+  /** property to explore */
+  property: IProperty
+  /** the current level of the hierarchy */
+  level?: number
+  /** the maximum depth to expanded nested types. -1 = all (default), 0 = no expansion */
+  maxDepth?: number
+  /** open all nodes immediately? */
+  openAll?: boolean
+}
+
+/**
+ * Show required status if a property is required
+ * @param property to describe
+ * @constructor
+ */
+export const ExplorePropertyRequired: FC<ExplorePropertyProps> = ({
+  property,
+}) => {
+  const tip = `${property.fullName} is required`
+  return (
+    <TipIcon
+      show={property.required}
+      icon="Check"
+      tip={tip}
+      title="required property"
+    />
+  )
+}
```

Comment:
Please see below for original comment

Reply:
I'm not sure what you're suggesting here other than an inline return. I prefer to leave TSX/JSX React statements declarative without embedded logic. I find them much more readable, and debugging support far superior.

Reply:
also a fair point. This can be considered resolved